1.9.0.CR1 announcement (#788)
debezium/debezium.github.io: _posts/2022-03-25-debezium-1-9-cr1-released.adoc

---
layout: post
title:  Debezium 1.9.0.CR1 Released
date:   2022-03-25
tags: [ releases, mysql, postgres, sqlserver, cassandra, oracle, db2, vitess, outbox ]
author: ccranfor
---

I am happy to announce the release of Debezium *1.9.0.CR1*!

> This release focuses primarily on bugfixes but includes support for Cassandra 4!

Reviewer: ```suggestion
Besides a range of bugfixes, this release brings the long-awaited support for Apache Cassandra 4!
```

Overall, https://issues.redhat.com/issues/?jql=project%20%3D%20DBZ%20AND%20fixVersion%20%3D%201.9.0.CR1%20ORDER%20BY%20component%20ASC[52 issues] have been fixed for this release.

Let's take a closer look at both the Cassandra 3 changes & Cassandra 4 support.

+++<!-- more -->+++

== Cassandra 3 changes & Cassandra 4 support

=== Cassandra 3 breaking changes

> For users who need to use Cassandra 3, the Maven coordinates have changed slightly in this release.

Reviewer: ```suggestion
For users who need to use Cassandra 3, the Maven coordinates of the (incubating) connector have changed slightly in this release.
```

  <artifactId>debezium-connector-cassandra-3</artifactId>
  <version>1.9.0.CR1</version>
</dependency>
```

There is one additional user-facing change that this release introduces, which is a shift in the Cassandra driver.
The connection configuration is no longer provided directly in the connector properties file but instead must be supplied using a separate `application.conf` file.
You can find a full reference on the driver's configuration https://docs.datastax.com/en/developer/java-driver/4.2/manual/core/configuration/reference/[here] and below is an example:

```
datastax-java-driver {
  basic {
    request.timeout = 20 seconds
    contact-points = [ "spark-master-1:9042" ]
    load-balancing-policy {
      local-datacenter = "dc1"
    }
  }
  advanced {
    auth-provider {
      class = PlainTextAuthProvider
      username = user
      password = pass
    }
    ssl-engine-factory {
     ...
    }
  }
}
```

In order for the Debezium connector to read/use this new application configuration file, it must be set in the connector properties file as follows:

```properties
cassandra.driver.config.file=/path/to/application/configuration.conf
```

=== Cassandra 4 support

> For new users and those looking to upgrade to Cassandra 4, the Maven coordinates for the new artifact are:

Reviewer: ```suggestion
For new users and those looking to upgrade to Cassandra 4, the Maven coordinates for the new connector artifact are:
```

```xml
<dependency>
  <groupId>io.debezium</groupId>
  <artifactId>debezium-connectr-cassandra-4</artifactId>
  <version>1.9.0.CR1</version>
</dependency>
```

We introduced a new artifact rather than a user configurable toggle as this allows both code bases to diverge as needed.

> This allows both Cassandra 3 and 4 to be refined as needed as we move forward to building Cassandra 4 with Java 11 as a baseline.

Reviewer: ```suggestion
This allows both the Cassandra 3 and 4 connectors to be refined as needed, as we move forward to building the Cassandra 4 connector with Java 11 as a baseline.
```

* Various MySQL connector improvements (https://issues.redhat.com/browse/DBZ-4758[DBZ-4758], https://issues.redhat.com/browse/DBZ-4787[DBZ-4787])

Please refer to the link:/releases/1.9/release-notes#release-1.9.0-cr1[release notes] to learn more about these and further fixes in this release.

As always, a big thank you to everyone contributing to this release:

https://github.com/roldanbob[Bob Roldan],
https://github.com/Naros[Chris Cranford],
https://github.com/clement-loiselet-talend[Clément Loiselet],
https://github.com/EthanZ328[Ethan Zou],
https://github.com/gunnarmorling[Gunnar Morling],
https://github.com/harveyyue[Harvey Yue],
https://github.com/jcechace[Jakub Cechacek],
https://github.com/novotnyJiri[Jiri Novotny],
https://github.com/jpechane[Jiri Pechanec],
https://github.com/josetesan[Jose Luis Sánchez],
https://github.com/jribera-sugarcrm[Josh Ribera],
https://github.com/kgalieva[Katerina Galieva],
https://github.com/nathan-smit-1[Nathan Smit],
https://github.com/zalmane[Oren Elias],
https://github.com/roldanbob[Robert Roldan],
https://github.com/morozov[Sergei Morozov],
https://github.com/smiklosovic[Stefan Miklosovic],

> https://github.com/vjuranek[Vojtech Juranek],

Reviewer: ```suggestion
```

Vojta is there twice.
